packages/fetchai/connections/gym/connection.py

```diff
@@ -220,15 +222,18 @@ def stop(self) -> None:
 
     @classmethod
     def from_config(
-        cls, address: Address, configuration: ConnectionConfig
+        cls, configuration: ConnectionConfig, identity: Identity, cryptos: CryptoStore
     ) -> "Connection":
         """
         Get the Gym connection from the connection configuration.
 
-        :param address: the address of the agent.
-        :param configuration: the connection configuration object.
+        :param configuration: the connection configuration.
+        :param identity: the identity object.
+        :param cryptos: object to access the connection crypto objects.
         :return: the connection object
         """
         gym_env_package = cast(str, configuration.config.get("env"))
         gym_env = locate(gym_env_package)
-        return GymConnection(gym_env(), address=address, configuration=configuration,)
+        return GymConnection(
+            gym_env(), configuration=configuration, identity=identity, cryptos=cryptos
+        )
```

Comment:
Why don't we make from_config a class method in the ABC instead? Would save us this awkward method. The configuration is now available in the constructor of the connection, so anything can be initialized there...

Reply:
yes that would be helpful.